### PR TITLE
Magento 2 4 0 Beta Release Notes

### PR DESCRIPTION
**Pull request by jfrontain**
## Purpose of this pull request

This pull request (PR) adds the Beta version of the 2.4.0 release notes for Beta deployment.

See internal ticket MC-34097 for details and link to staging build.

whatsnew
Added [release notes highlights](https://devdocs.magento.com/guides/v2.4/release-notes/bk-release-notes.html) to support the 2.4.0 beta release.

### CODE REVIEW

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 1]
```diff
@@ -0,0 +1,1168 @@
...
+
+## Highlights
+
+Look for the following highlights in this release:
+
+### Substantial security enhancements
+
+This release includes the following security enhancements:
+
+#### Over 30 security enhancements that help close remote code execution (RCE) and cross-site scripting (XSS) vulnerabilities
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:38]: Is this heading necessary or could it be a bullet with the following paragraphs in lines 40-42 indented underneath?

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 2]
```diff
@@ -0,0 +1,1168 @@
...
+
+No confirmed attacks related to these issues have occurred to date. However, certain vulnerabilities can potentially be exploited to access customer information or take over administrator sessions. Most of these issues require that an attacker first obtains access to the Admin. As a result, we remind you to take all necessary steps to protect your Admin, including but not limited to these efforts: IP whitelisting, [two-factor authentication]({{page.baseurl}}/security/two-factor-authentication.html), use of a VPN, the use of a unique location rather than `/admin`, and good password hygiene. See Security updates available for Magento for a discussion of these fixed issues.
+
+With the Magento 2.3.4 release, we changed how we describe these security issues.  Individual issues are no longer described in the Magento Security Center. Instead, these issues are documented in an Adobe Security bulletin.
+
+#### Security enhancements and fixes to core code
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:44]: Is this heading necessary? It seems like all security items should be consolidated into one list under a single heading, which already exists on line 34.

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 3]
```diff
@@ -0,0 +1,1168 @@
...
+
+*  **Template filter strict mode is now enabled by default**. Magento components (including CMS pages and blocks) that use the template filter in legacy mode can be vulnerable to remote code execution (RCE). Enabling strict mode by default ensures that RCE attacks cannot be deliberately enabled. <!--- MC-22982-->
+
+*  **Data rendering for UI data providers is now disabled by default**. This removes an opportunity for malicious users to execute arbitrary JavaScript. <!--- MC-17356-->
+
+*  **Content Security Policy (CSP) improvements**. `SecureHtmlRenderer` has been added to the Magento Framework and is available in `.phtml` templates to support the whitelisting of inline `style` and `script` tags. Inline scripts and styles are not typically permitted with the default CSP configuration. Although most core template files have been updated, some pages may display CSP violations.
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:54]: Add a link to the the Content Security Policy topic.

https://devdocs.magento.com/guides/v2.4/extension-dev-guide/security/content-security-policies.html#whitelist-an-inline-script-or-style

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 4]
```diff
@@ -0,0 +1,1168 @@
...
+
+*  **Removal of the MySQL catalog search engine**. The MySQL search engine has been removed from Magento 2.4.0 and replaced as the default search engine with Elasticsearch. Elasticsearch provides superior search capabilities as well as catalog performance optimizations.  All merchants must have Elasticsearch to install and deploy Magento 2.4.0. See [Check the catalog search engine]({{page.baseurl}}/comp-mgr/prereq/prereq-elasticsearch.html).
+
+*  **Migration of dependencies on Zend Framework to the [Laminas project](https://getlaminas.org/about/foundation)** to reflect the transitioning of Zend Framework to the Linux Foundation’s Laminas Project. Zend Framework has been deprecated. See the [Migration of Zend Framework to the Laminas Project](https://community.magento.com/t5/Magento-DevBlog/Migration-of-Zend-Framework-to-the-Laminas-Project/ba-p/443251)  DevBlog post.
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:73]:
```suggestion
*  **Migration of dependencies on Zend Framework to the [Laminas project](https://getlaminas.org/about/foundation)** to reflect the transitioning of Zend Framework to the Linux Foundation’s Laminas Project. Zend Framework has been deprecated. See the [Migration of Zend Framework to the Laminas Project](https://community.magento.com/t5/Magento-DevBlog/Migration-of-Zend-Framework-to-the-Laminas-Project/ba-p/443251) DevBlog post.
```

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 5]
```diff
@@ -0,0 +1,1168 @@
...
+
+*  **Removal of the core integration of the Signifyd fraud protection code**. This core feature is no longer supported. Merchants should migrate to the [Signifyd Fraud & Chargeback Protection extension](https://marketplace.magento.com/signifyd-module-connect.html) that is available on the Magento Marketplace.
+
+*  **MariaDB 10.4 support**. Support for MySQL 8.0 provides the opportunity for merchants to deploy MariaDB 10.4 with Magento. Although merchants can still use MariaDB 10.2 with Magento 2.4.0, we recommend upgrading to MariaDB 10.4 for improved performance and reliability. MariaDB 10.0 and 10.1 are no longer supported (as a result of removing support for MySQL 5.6 in this release).
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:77]: I think this bullet should be moved immediately below the MySQL bullet on line 69.

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 6]
```diff
@@ -0,0 +1,1168 @@
...
+
+*  **PHP 7.4 support introduced and PHP 7.1 and 7.2 deprecated**. Magento 2.4.0 introduces support for PHP 7.4. All testing of 2.4.0 has been done on PHP 7.4. You can still install Magento 2.4.x with PHP 7.3, but we recommend using PHP 7.4.
+
+*  **Support for PHPUnit 9.x and deprecation of PHPUnit 6.5**. PHP 7.4 requires the use of the latest PHPUnit testing framework, which is PHPUnit 9.x. Magento Marketplace extension vendors must confirm that all new extension versions are compatible with PHP 7.4 and that all  unit and integration tests have been configured to be run with PHPUnit 9.
+
+*  **Elasticsearch 7.x support**. The latest Elasticsearch 7.x version is now the default catalog search engine for Magento Commerce and Open Source. Elasticsearch version 2.x code has been removed, and Elasticsearch version 5.x has been deprecated. Elasticsearch v6.8 is still supported in Magento 2.4.0 (and 2.3.x). See [Elasticsearch]({{page.baseurl}}/install-gde/prereq/elasticsearch.html).
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:67]: Should we note that you cannot install, or upgrade to, 2.4.0 if you do not have ES installed?

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 7]
```diff
@@ -0,0 +1,1168 @@
...
+
+*  **Support for partial-word search for Elasticsearch (new default search engine)**. Elasticsearch now supports the use of  partial words in search terms for  product names and SKUs when using quick search. This capability was supported by the MySQL search engine, which has been deprecated and replaced by Elasticsearch in this release.
+
+*  **PayPal JavaScript SDK upgrade**. We’ve migrated the PayPal Express Checkout integration to the latest PayPal JavaScript SDK, an SDK that  automatically collects and passes needed risk parameters to PayPal. The behavior of the PayPal Express Checkout payment method remains unchanged. However, upgrading this SDK to the latest version let merchants access the latest features and security enhancements. <!--- MC-30962-->
+
+*  **Deprecation and removal of the Web Set Up Wizard**.
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:91]: Consider adding a sentence letting devs know they must use the command-line to install or upgrade Magento in 2.4.0 and link to the corresponding topics.

- https://devdocs.magento.com/guides/v2.4/install-gde/install/cli/install-cli.html
- https://devdocs.magento.com/guides/v2.4/comp-mgr/cli/cli-upgrade.html

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 8]
```diff
@@ -0,0 +1,1168 @@
...
+
+*  **Composer update plugin**. Composer plugin streamlines the  upgrade process by resolving changes that must be made to the root project `composer.json` file before updating to a new Magento product requirement. This plug-in protects against overwriting customizations. See [Upgrade using the Magento composer root plugin](https://devdocs.magento.com/guides/v2.3/comp-mgr/cli/upgrade-with-plugin.html).
```
Comment (jeff-matthews) [inline src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:93]:
```suggestion
*  **Composer update plugin**. Composer plugin streamlines the  upgrade process by resolving changes that must be made to the root project `composer.json` file before updating to a new Magento product requirement. This plug-in protects against overwriting customizations. See [Upgrade using the Magento composer root plugin](https://devdocs.magento.com/guides/v2.4/comp-mgr/cli/cli-upgrade.html).
```

Comment (jeff-matthews) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:93]: I removed the `upgrade-with-plugin.html` file in the 2.4.0-develop branch and consolidated the info with the top-level `cli-upgrade.html` file.

File: src/_data/toc/release-notes.yml [hunk 9]
```diff
@@ -3,6 +3,20 @@ pages:
     - label: Introduction
       url: /release-notes/bk-release-notes.html
 
+    - label: 2.4 Release Notes
+      include_versions: ["2.4"]
+      children:
+
+        - label: Component Status
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/_data/toc/release-notes.yml]: done

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 10]
```diff
@@ -0,0 +1,1168 @@
+---
+group: release-notes
+title: Magento Commerce 2.4.0 Release Notes
+---
+
+Magento Commerce 2.4.0 offers significant platform upgrades, substantial security changes, and performance improvements.
+
+This release includes all the improvements to core quality that were included in Magento 2.3.5,  over 100 new fixes to core code, and 30 security enhancements. It includes the resolution of  226 GitHub issues by our community members. These community contributions range from minor clean-up of core code to significant enhancements in Inventory Management and GraphQL.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: done -- thanks for catching this!

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 11]
```diff
@@ -0,0 +1,1168 @@
...
+
+{:.bs-callout-info}
+
+Quarterly releases may contain backward-incompatible changes (BIC). Magento 2.4.0 contains minor backward-incompatible changes. To review minor backward-incompatible changes, see [BIC reference]({{page.baseurl}}/release-notes/backward-incompatible-changes/reference.html). (Major backward-incompatible issues are described in [BIC highlights]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html). Not all releases introduce major BICs.)
+
+{:.bs-callout-info}
+
+The name of the security-only patch for this quarter is 2.3.5-p2. Future releases will follow the typical package naming conventions for full-release  and security packages.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: done

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 12]
```diff
@@ -0,0 +1,1168 @@
...
+
+## Security-only patch available
+
+Merchants can now install time-sensitive security fixes without applying the hundreds of functional fixes and enhancements that a full quarterly release (for example, Magento 2.3.5-p2) provides. Patch 2.3.5.2 (Composer package 2.3.5-p2) is a security-only patch that provides fixes for vulnerabilities that have been identified in our previous quarterly release, Magento 2.3.5. All hot fixes that were applied to the 2.3.5 release are included in this security-only patch. (A *hot fix* provides a fix to a released version of Magento that addresses a specific problem or bug.)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: done

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 2 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: removed a redundant header and trimmed this a bit -- thanks

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 3 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: added

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 4 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: done

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 6 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: best to be really clear on this -- thanks -- added sentence

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 8 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: changed link -- thanks

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 7 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: added sentence

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 5 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: moved bullet

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md (hunk 1 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md:36]: streamlined this section

File: src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md [hunk 13]
```diff
@@ -0,0 +1,1168 @@
...
+
+### Substantial security enhancements
+
+This release includes the following security enhancements:
+
+#### Over 30 security enhancements that help close remote code execution (RCE) and cross-site scripting (XSS) vulnerabilities
+
+No confirmed attacks related to these issues have occurred to date. However, certain vulnerabilities can potentially be exploited to access customer information or take over administrator sessions. Most of these issues require that an attacker first obtains access to the Admin. As a result, we remind you to take all necessary steps to protect your Admin, including but not limited to these efforts: IP whitelisting, [two-factor authentication]({{page.baseurl}}/security/two-factor-authentication.html), use of a VPN, the use of a unique location rather than `/admin`, and good password hygiene. See Security updates available for Magento for a discussion of these fixed issues.
+
+With the Magento 2.3.4 release, we changed how we describe these security issues.  Individual issues are no longer described in the Magento Security Center. Instead, these issues are documented in an Adobe Security bulletin.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/release-notes-2-4-0-commerce.md]: I think I removed this but might have edited it differently. Will double-check and edit this part of the intro after Beta

**Comment by jfrontain** [comment]
running tests

**Pull request merged**